### PR TITLE
test: Allow ensemble to create the final response even if some of the outputs are not created

### PR DESCRIPTION
**Pull request by kthui**
#### What does the PR do?
Add test when the composing models of an ensemble do not provide all outputs on the model configuration. The inference should complete normally and return all outputs that are provided.

#### Checklist
- [x] PR title reflects the change and is of format `<commit_type>: <Title>`
- [x] Changes are described in the pull request.
- [x] Related issues are referenced.
- [x] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [x] Verified copyright is correct on all changed files.
- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [x] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [ ] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [x] test

#### Related PRs:
https://github.com/triton-inference-server/core/pull/399

#### Where should the reviewer start?
N/A

#### Test plan:
N/A

- CI Pipeline ID: 19348789

#### Caveats:
N/A

#### Background
N/A

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
N/A

### CODE REVIEW

**Review by GuanLuo — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]

File: qa/L0_simple_ensemble/ensemble_test.py [hunk 1]
```diff
@@ -177,6 +178,41 @@ def callback(start_time, result, error):
         if expected_flags != response_flags:
             self.assertTrue(False, "unexpeced sequence flags mismatch error")
 
+    def test_ensemble_partial_add_sub(self):
```
Comment (rmccorm4) [inline qa/L0_simple_ensemble/ensemble_test.py:181]: I think we should have more testing in general to programatically track expected behavior of certain edge cases, such as:
1. What if a composing model doesn't produce the outputs necessary to feed inputs of the next step?
2. In an ensemble with outputs from 2 parallel steps, what if 1 step has outputs and the other step has no outputs?
3. What if an ensemble doesn't produce any of its outputs?

These aren't immediately clear to me from scanning the code what would happen, so I think it would be good to enforce expected behavior via testing, and be able to catch if the behavior subtlely changes from future ensemble code changes

Comment (rmccorm4) [reply qa/L0_simple_ensemble/ensemble_test.py:181]: @kthui
